File: Day 29 DSA (404)/code.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def recurse(node):
            if not node:
                return 0
            add = 0
            if node.left and not node.left.left and not node.left.right:
                add += node.left.val
            return add + recurse(node.left) + recurse(node.right)
        
        return recurse(root)

File: Day 38 DSA (700)/Code.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* searchBST(TreeNode* root, int val) {
        if (!root)
            return nullptr;
        if (!root->left && !root->right)
            return root->val == val ? root : nullptr;
        if(root->val == val){
            return root;
        }
        if(root->val < val && root->right)
        {
            return searchBST(root->right, val);
        }
        if(root->val > val && root->left)
        {
            return searchBST(root->left, val);
        }
        return nullptr;
    }
};
